cli/pcluster/utils.py

```diff
@@ -131,6 +131,23 @@ def _get_json_from_s3(region, file_name):
     return json.loads(file_contents)
 
 
+def get_supported_efa_instances(region):
+    """
+    Get a json object containing the instances supported by efa.
+
+    :param region: AWS Region
+    :param instance_type: the instance type to search for.
+    :return: json object containing the instances supported by efa
+    or an empty object if unable to parse/get the instance list file
+    """
+    try:
+        instances = _get_json_from_s3(region, "instances/efa_instances.json")
```

Comment:
would it make sense to add efa details to the current instances.json file or have a more general purpose json file we can use to dynamically enable support for different features so that we don't need to pull a single file per feature?

Reply:
yes, let's talk about the best way to do this

Reply:
I've revised the creation of this file in https://github.com/aws/aws-parallelcluster/pull/1067

Reply:
Will you change this in order to call get_supported_instances? Also shall we fetch also schedulers and oses?

Reply:
Yep 👍

Changed to `get_supported_instances()` everywhere.